src/components/webview/sourceControl.ts

```diff
@@ -90,4 +97,29 @@ export class SourceControlService {
 		this.__messageBus.publish({ kind: MessageKind.afterIssueCreated });
 		return result.data;
 	}
+
+	async listPR() {
+		const repositoryPath = this.__repositoryService.getRepositoryPath();
+
+		if (!repositoryPath) {
+			throw new NotFoundRepositoryPath();
+		}
+
+		const userId = this.__userAccountStorage.getUserAccount();
+
+		if (!userId) {
+			throw new NotFoundIntuitaAccount();
+		}
+
+		const result = await axios.get<ListPRResponse>(
+			'https://telemetry.intuita.io/sourceControl/github/pulls',
+		);
+
+		return result.data;
+	}
+
+	async getPRForBranch(branchName: string) {
+		const PRList = await this.listPR();
+		return PRList.find((pr) => pr.head.ref === branchName);
```

Comment:
I'd consider using `?? null` at the end (`undefined` makes little sense for a list that cannot contain null values)